pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/utilities/SecuredHDFS.java

```diff
@@ -53,6 +53,10 @@
     public static void verifyToken(ProtocolData protData, ServletContext context) {
         try {
             if (UserGroupInformation.isSecurityEnabled()) {
+                UserGroupInformation loginUser = UserGroupInformation.getLoginUser();
+                if (!loginUser.hasKerberosCredentials()) {
+                    SecureLogin.login();
```

Comment:
Instead of `SecureLogin` we can just directly use `UserGroupInformation.getLoginUser().reloginFromKeytab()` seems more lightweight since it leverages all the kerberos information that's already in memory